### DISCUSSION TITLE
How can I type response data from a invoke service (v5.x)

### DISCUSSION THREAD

**Discussion by goldenratio** [statelyai/xstate#5081] [Q&A] [answered]
Hey,

I would like to have type completion for invoke (actor) `onDone`'s output data. At the moment I am getting `any`, how can I add `type` to this.

Note: I provide actors to the machine externally with `.provide()` method.

![Screenshot from 2024-09-18 11-08-09](https://github.com/user-attachments/assets/785800da-2b45-4438-85f6-d83d2ad036a6)

```
const actor = createActor(gameplayMachine.provide({
    actions: {},
    actors: {
      getGameJoinData: fromPromise(async () => {
        await delay(2000);
        const data = await service.connect();
        return { ...data };
      })
    },
    guards: {},
  }), {
    inspect
  });
```
Thanks for your help..

**Comment by davidkpiano** [comment, marked as answer]
You can do so by using the `setup(…)` function and providing the actor. See here: https://stately.ai/docs/invoke#invoke-and-typescript

**Reply by goldenratio** [reply]
thanks
